processing/src/main/java/io/druid/query/aggregation/post/DoubleGreatestPostAggregator.java

```diff
@@ -81,13 +77,15 @@ public Comparator getComparator()
   public Object compute(Map<String, Object> values)
   {
     Iterator<PostAggregator> fieldsIter = fields.iterator();
-    double retVal = Double.NEGATIVE_INFINITY;
-    if (fieldsIter.hasNext()) {
-      retVal = ((Number) fieldsIter.next().compute(values)).doubleValue();
-      while (fieldsIter.hasNext()) {
-        double other = ((Number) fieldsIter.next().compute(values)).doubleValue();
-        if (other > retVal) {
-          retVal = other;
+    Double retVal = NullHandling.replaceWithDefault() ? Double.NEGATIVE_INFINITY : null;
+    while (fieldsIter.hasNext()) {
+      Number nextVal = ((Number) fieldsIter.next().compute(values));
```

Comment:
It looks that we need specialized `PostAggregator` for primitive types and a new method `isNull()` to avoid boxing/unboxing. Would you file an issue for this?

Reply:
Why? Does performance matter here?

Reply:
I think so. `compute()` method is called per row on brokers.